### PR TITLE
cli: discern unsupported plugin type.

### PR DESCRIPTION
**Pull request by Txiaozhe**
When user input one unsupported plugin type, it should be discerned.
```shell
➜  pipcook-project pipcook plugin-dev -t unsupported-type
Unsupported plugin type: unsupported-type.
The type of plugin must be one of: 
{ dataAccess, dataCollect, dataProcess, modelDeploy, modelEvaluate, modelLoad, modelTrain }
➜  pipcook-project 
```

### CODE REVIEW

File: packages/pipcook-cli/lib/devPlugin.js [hunk 1]
```diff
@@ -15,12 +15,20 @@ const devPlugin = (cmdObj) => {
     return;
   }
 
-  if (!projectName) {
-    projectName = 'template-plugin';
-  }
-
   let dirname;
   try {
+    const typeFiles = fse.readdirSync(path.join(__dirname, '../assets/pluginPackage/src'));
```
Comment (yorkie) [inline packages/pipcook-cli/lib/devPlugin.js]: All plugins are exported at https://github.com/alibaba/pipcook/blob/master/packages/pipcook-core/src/constants/plugins.ts#L1-L8.

File: packages/pipcook-cli/lib/devPlugin.js [hunk 2]
```diff
@@ -15,12 +15,20 @@ const devPlugin = (cmdObj) => {
...
+    const allowedTypes = typeFiles.map(filename => filename.replace(/(.*\/)*([^.]+).*/ig,"$2"));
```
Comment (yorkie) [inline packages/pipcook-cli/lib/devPlugin.js]: Wrapping this in a function with a meaningful name would be better.

File: packages/pipcook-cli/lib/devPlugin.js [hunk 3]
```diff
@@ -15,12 +15,20 @@ const devPlugin = (cmdObj) => {
...
+    if (allowedTypes.indexOf(pluginType) < 0) {
+      console.log(`Unsupported plugin type: ${pluginType}.\n` +
```
Comment (yorkie) [inline packages/pipcook-cli/lib/devPlugin.js]: Use `console.warn`

Comment (yorkie) [reply packages/pipcook-cli/lib/devPlugin.js]: And may I ask you to split into multiple calls instead of `\n`?

File: packages/pipcook-cli/lib/devPlugin.js [hunk 4]
```diff
@@ -15,12 +15,20 @@ const devPlugin = (cmdObj) => {
...
+        `The type of plugin must be one of: \n{ ${allowedTypes.join(', ')} }`);
+      return;
+    }
+
+    if (!projectName) {
+      projectName = 'template-plugin';
```
Comment (yorkie) [inline packages/pipcook-cli/lib/devPlugin.js]: This should be reverted because it seems unrelated?

**Comment by Txiaozhe** [comment]
The final effect:

File: packages/pipcook-cli/lib/devPlugin.js [hunk 5]
```diff
@@ -15,20 +17,19 @@ const devPlugin = (cmdObj) => {
...
-    const allowedTypes = typeFiles.map(filename => filename.replace(/(.*\/)*([^.]+).*/ig,"$2"));
-    if (allowedTypes.indexOf(pluginType) < 0) {
-      console.log(`Unsupported plugin type: ${pluginType}.\n` +
-        `The type of plugin must be one of: \n{ ${allowedTypes.join(', ')} }`);
-      return;
-    }
+  if (constants.PLUGINS.indexOf(pluginType) < 0) {
+    console.warn(chalk.red(`Unsupported plugin type: "${pluginType}"`));
+    console.warn(chalk.yellow('The type of plugin must be one of: '));
+    console.warn(chalk.yellow(`{ \n  ${constants.PLUGINS.join(',\n  ')}\n}`));
```
Comment (yorkie) [inline packages/pipcook-cli/lib/devPlugin.js]: Just print them in list

File: packages/pipcook-cli/lib/devPlugin.js [hunk 6]
```diff
@@ -15,6 +17,11 @@ const devPlugin = (cmdObj) => {
     return;
   }
 
+  if (constants.PLUGINS.indexOf(pluginType) < 0) {
+    console.warn(chalk.red(`Unsupported plugin type: "${pluginType}", it must be one of: \n${constants.PLUGINS.join(',\n')}`));
```
Comment (yorkie) [inline packages/pipcook-cli/lib/devPlugin.js]: This line is too long, namely over 80 chars.

Comment (Txiaozhe) [reply packages/pipcook-cli/lib/devPlugin.js]: Yeah, and many lines in this project are over 80 chars like this, we need a lint for standardized the code:)

Comment (yorkie) [reply packages/pipcook-cli/lib/devPlugin.js]: Ok, resolved.
